File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Cloudfw/AlibabaCloud/SDK/ROS/CDK/Cloudfw/Datasource/IAddressBooks.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Cloudfw.Datasource
{
    /// <summary>Represents a `AddressBooks`.</summary>
    [JsiiInterface(nativeType: typeof(IAddressBooks), fullyQualifiedName: "@alicloud/ros-cdk-cloudfw.datasource.IAddressBooks")]
    public interface IAddressBooks : AlibabaCloud.SDK.ROS.CDK.Core.IResource
    {
        /// <summary>Attribute AddressBooks: The list of address books.</summary>
        [JsiiProperty(name: "attrAddressBooks", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrAddressBooks
        {
            get;
        }

        /// <summary>Attribute GroupUuids: The list of group uuids.</summary>
        [JsiiProperty(name: "attrGroupUuids", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrGroupUuids
        {
            get;
        }

        [JsiiProperty(name: "props", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-cloudfw.datasource.AddressBooksProps\"}")]
        AlibabaCloud.SDK.ROS.CDK.Cloudfw.Datasource.IAddressBooksProps Props
        {
            get;
        }

        /// <summary>Represents a `AddressBooks`.</summary>
        [JsiiTypeProxy(nativeType: typeof(IAddressBooks), fullyQualifiedName: "@alicloud/ros-cdk-cloudfw.datasource.IAddressBooks")]
        internal sealed class _Proxy : DeputyBase, AlibabaCloud.SDK.ROS.CDK.Cloudfw.Datasource.IAddressBooks
        {
            private _Proxy(ByRefValue reference): base(reference)
            {
            }

            /// <summary>Attribute AddressBooks: The list of address books.</summary>
            [JsiiProperty(name: "attrAddressBooks", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrAddressBooks
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute GroupUuids: The list of group uuids.</summary>
            [JsiiProperty(name: "attrGroupUuids", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrGroupUuids
            {
                get => GetInstanceProperty<object>()!;
            }

            [JsiiProperty(name: "props", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-cloudfw.datasource.AddressBooksProps\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Cloudfw.Datasource.IAddressBooksProps Props
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Cloudfw.Datasource.IAddressBooksProps>()!;
            }

            /// <summary>The environment this resource belongs to.</summary>
            /// <remarks>
            /// For resources that are created and managed by the CDK
            /// (generally, those created by creating new class instances like Role, Bucket, etc.),
            /// this is always the same as the environment of the stack they belong to;
            /// however, for imported resources
            /// (those obtained from static methods like fromRoleArn, fromBucketName, etc.),
            /// that might be different than the stack they were imported into.
            /// </remarks>
            [JsiiProperty(name: "env", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.ResourceEnvironment\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.IResourceEnvironment Env
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.IResourceEnvironment>()!;
            }

            /// <summary>The stack in which this resource is defined.</summary>
            [JsiiProperty(name: "stack", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.Stack\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.Stack Stack
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.Stack>()!;
            }

            /// <summary>The construct tree node for this construct.</summary>
            [JsiiProperty(name: "node", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.ConstructNode\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode Node
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode>()!;
            }
        }
    }
}

File: multiple-languages/java/ros-cdk-cloudfw/src/main/java/com/aliyun/ros/cdk/cloudfw/datasource/IAddressBooks.java
package com.aliyun.ros.cdk.cloudfw.datasource;

/**
 * Represents a <code>AddressBooks</code>.
 */
@javax.annotation.Generated(value = "jsii-pacmak/1.85.0 (build 08ee592)", date = "2025-09-28T10:11:22.207Z")
@software.amazon.jsii.Jsii(module = com.aliyun.ros.cdk.cloudfw.$Module.class, fqn = "@alicloud/ros-cdk-cloudfw.datasource.IAddressBooks")
@software.amazon.jsii.Jsii.Proxy(IAddressBooks.Jsii$Proxy.class)
public interface IAddressBooks extends software.amazon.jsii.JsiiSerializable, com.aliyun.ros.cdk.core.IResource {

    /**
     * Attribute AddressBooks: The list of address books.
     */
    @org.jetbrains.annotations.NotNull java.lang.Object getAttrAddressBooks();

    /**
     * Attribute GroupUuids: The list of group uuids.
     */
    @org.jetbrains.annotations.NotNull java.lang.Object getAttrGroupUuids();

    @org.jetbrains.annotations.NotNull com.aliyun.ros.cdk.cloudfw.datasource.AddressBooksProps getProps();

    /**
     * A proxy class which represents a concrete javascript instance of this type.
     */
    @software.amazon.jsii.Internal
    final class Jsii$Proxy extends software.amazon.jsii.JsiiObject implements com.aliyun.ros.cdk.cloudfw.datasource.IAddressBooks.Jsii$Default {
        protected Jsii$Proxy(final software.amazon.jsii.JsiiObjectRef objRef) {
            super(objRef);
        }

        /**
         * The construct tree node for this construct.
         */
        @Override
        public final @org.jetbrains.annotations.NotNull com.aliyun.ros.cdk.core.ConstructNode getNode() {
            return software.amazon.jsii.Kernel.get(this, "node", software.amazon.jsii.NativeType.forClass(com.aliyun.ros.cdk.core.ConstructNode.class));
        }

        /**
         * The environment this resource belongs to.
         * <p>
         * For resources that are created and managed by the CDK
         * (generally, those created by creating new class instances like Role, Bucket, etc.),
         * this is always the same as the environment of the stack they belong to;
         * however, for imported resources
         * (those obtained from static methods like fromRoleArn, fromBucketName, etc.),
         * that might be different than the stack they were imported into.
         */
        @Override
        public final @org.jetbrains.annotations.NotNull com.aliyun.ros.cdk.core.IResourceEnvironment getEnv() {
            return software.amazon.jsii.Kernel.get(this, "env", software.amazon.jsii.NativeType.forClass(com.aliyun.ros.cdk.core.IResourceEnvironment.class));
        }

        /**
         * The stack in which this resource is defined.
         */
        @Override
        public final @org.jetbrains.annotations.NotNull com.aliyun.ros.cdk.core.Stack getStack() {
            return software.amazon.jsii.Kernel.get(this, "stack", software.amazon.jsii.NativeType.forClass(com.aliyun.ros.cdk.core.Stack.class));
        }

        /**
         * Attribute AddressBooks: The list of address books.
         */
        @Override
        public final @org.jetbrains.annotations.NotNull java.lang.Object getAttrAddressBooks() {
            return software.amazon.jsii.Kernel.get(this, "attrAddressBooks", software.amazon.jsii.NativeType.forClass(java.lang.Object.class));
        }

        /**
         * Attribute GroupUuids: The list of group uuids.
         */
        @Override
        public final @org.jetbrains.annotations.NotNull java.lang.Object getAttrGroupUuids() {
            return software.amazon.jsii.Kernel.get(this, "attrGroupUuids", software.amazon.jsii.NativeType.forClass(java.lang.Object.class));
        }

        @Override
        public final @org.jetbrains.annotations.NotNull com.aliyun.ros.cdk.cloudfw.datasource.AddressBooksProps getProps() {
            return software.amazon.jsii.Kernel.get(this, "props", software.amazon.jsii.NativeType.forClass(com.aliyun.ros.cdk.cloudfw.datasource.AddressBooksProps.class));
        }
    }

    /**
     * Internal default implementation for {@link IAddressBooks}.
     */
    @software.amazon.jsii.Internal
    interface Jsii$Default extends IAddressBooks, com.aliyun.ros.cdk.core.IResource.Jsii$Default {

        /**
         * The construct tree node for this construct.
         */
        @Override
        default @org.jetbrains.annotations.NotNull com.aliyun.ros.cdk.core.ConstructNode getNode() {
            return software.amazon.jsii.Kernel.get(this, "node", software.amazon.jsii.NativeType.forClass(com.aliyun.ros.cdk.core.ConstructNode.class));
        }

        /**
         * The environment this resource belongs to.
         * <p>
         * For resources that are created and managed by the CDK
         * (generally, those created by creating new class instances like Role, Bucket, etc.),
         * this is always the same as the environment of the stack they belong to;
         * however, for imported resources
         * (those obtained from static methods like fromRoleArn, fromBucketName, etc.),
         * that might be different than the stack they were imported into.
         */
        @Override
        default @org.jetbrains.annotations.NotNull com.aliyun.ros.cdk.core.IResourceEnvironment getEnv() {
            return software.amazon.jsii.Kernel.get(this, "env", software.amazon.jsii.NativeType.forClass(com.aliyun.ros.cdk.core.IResourceEnvironment.class));
        }

        /**
         * The stack in which this resource is defined.
         */
        @Override
        default @org.jetbrains.annotations.NotNull com.aliyun.ros.cdk.core.Stack getStack() {
            return software.amazon.jsii.Kernel.get(this, "stack", software.amazon.jsii.NativeType.forClass(com.aliyun.ros.cdk.core.Stack.class));
        }

        /**
         * Attribute AddressBooks: The list of address books.
         */
        @Override
        default @org.jetbrains.annotations.NotNull java.lang.Object getAttrAddressBooks() {
            return software.amazon.jsii.Kernel.get(this, "attrAddressBooks", software.amazon.jsii.NativeType.forClass(java.lang.Object.class));
        }

        /**
         * Attribute GroupUuids: The list of group uuids.
         */
        @Override
        default @org.jetbrains.annotations.NotNull java.lang.Object getAttrGroupUuids() {
            return software.amazon.jsii.Kernel.get(this, "attrGroupUuids", software.amazon.jsii.NativeType.forClass(java.lang.Object.class));
        }

        @Override
        default @org.jetbrains.annotations.NotNull com.aliyun.ros.cdk.cloudfw.datasource.AddressBooksProps getProps() {
            return software.amazon.jsii.Kernel.get(this, "props", software.amazon.jsii.NativeType.forClass(com.aliyun.ros.cdk.cloudfw.datasource.AddressBooksProps.class));
        }
    }
}
